client/src/PlayerLanding/AddPlayer.tsx

```diff
@@ -44,22 +44,21 @@ function AddPlayer() {
         </Callout.Text>
       </Callout.Root>
       <Flex direction="column" gap="2" className="p-2">
-        <label htmlFor="name-input">NAME:</label>
-        <TextField.Input
-          autoFocus
-          id="name-input"
-          placeholder="Player name..."
-          value={name}
-          onChange={(event) => setName(event.currentTarget.value)}
-          onKeyDown={async (event) => {
-            if (event.key === "Enter") {
-              handleSubmit();
-              // Otherwise it will also select in the dialog
-              event.stopPropagation();
-              event.preventDefault();
-            }
+        <form
+          onSubmit={(e) => {
+            e.preventDefault();
+            handleSubmit();
           }}
-        />
+        >
+          <label htmlFor="name-input">NAME:</label>
+          <TextField.Input
+            autoFocus
+            id="name-input"
+            placeholder="Player name..."
+            value={name}
+            onChange={(event) => setName(event.target.value)}
```

Comment:
Thinking the Radix component is making it so that currentTarget doesn't necessarily attach to the input, but that's just a guess right now.